### PR TITLE
[CORL-2124] Accessibility Improvements

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- Full Accessibility makeover for the comment stream

## How do I test this PR?
- Use a screenreader to navigate and use the stream with live traffic

### CODE REVIEW

File: src/core/client/stream/tabs/Comments/Stream/PostCommentForm/PostCommentFormContainer.tsx [hunk 1]
```diff
@@ -115,6 +117,23 @@ export const PostCommentFormContainer: FunctionComponent<Props> = ({
...
+    () => (props) => (
+      <section
+        aria-label={
+          isRatingsAndReviews
+            ? "Submit a Review or Ask a Question"
+            : isQA
+            ? "Post a Question"
+            : "Post a Comment"
+        }
+      >
```
Comment (cvle) [inline src/core/client/stream/tabs/Comments/Stream/PostCommentForm/PostCommentFormContainer.tsx:130]: this

File: src/core/client/stream/tabs/Comments/Stream/FeaturedComments/FeaturedCommentContainer.tsx [hunk 2]
```diff
@@ -59,10 +66,16 @@ const FeaturedCommentContainer: FunctionComponent<Props> = (props) => {
...
+      <article
         className={cn(CLASSES.featuredComment.$root, styles.root)}
         data-testid={`featuredComment-${comment.id}`}
+        aria-labelledby={`featuredComment-${comment.id}-label`}
       >
+        <Hidden id={`featuredComment-${comment.id}-label`}>
+          Featured Comment from {comment.author?.username || "Deleted User"}{" "}
+          {` `}
+          <RelativeTime date={comment.createdAt} />
+        </Hidden>
```
Comment (cvle) [inline src/core/client/stream/tabs/Comments/Stream/FeaturedComments/FeaturedCommentContainer.tsx:78]: This

File: src/core/client/stream/tabs/Profile/MyComments/HistoryComment.tsx [hunk 3]
```diff
@@ -35,8 +35,14 @@ const HistoryComment: FunctionComponent<HistoryCommentProps> = (props) => {
...
       className={cn(styles.root, CLASSES.myComment.$root)}
       data-testid={`historyComment-${props.id}`}
+      container="article"
+      aria-labelledby={`historyComment-${props.id}-label`}
     >
       <div>
+        <Hidden id={`historyComment-${props.id}-label`}>
+          Comment <RelativeTime date={props.createdAt} />{" "}
+          {props.story.metadata && `on ${props.story.metadata.title}`}
+        </Hidden>
```
Comment (cvle) [inline src/core/client/stream/tabs/Profile/MyComments/HistoryComment.tsx:45]: This

File: src/core/client/stream/tabs/Comments/Comment/CommentContainer.tsx [hunk 4]
```diff
@@ -355,11 +378,21 @@ export const CommentContainer: FunctionComponent<Props> = ({
...
       data-key-stop
     >
+      <Hidden id={`${commentElementID}-label`}>
+        {indentLevel && `Thread Level ${indentLevel}: `}
+        {ariaIsHighlighted && "Highlighted "}
+        {ariaIsAncestor && "Ancestor "}
+        {comment.parent ? "Reply" : isQA ? "Question" : "Comment"} from{" "}
+        {comment.author?.username || "Deleted User"} {` `}
+        <RelativeTime date={comment.createdAt} />
+      </Hidden>
```
Comment (cvle) [inline src/core/client/stream/tabs/Comments/Comment/CommentContainer.tsx:395]: This

File: src/core/client/stream/tabs/Comments/Stream/AnsweredCommentsTab/AnsweredCommentContainer.tsx [hunk 5]
```diff
@@ -68,10 +74,16 @@ const AnsweredCommentContainer: FunctionComponent<Props> = (props) => {
...
         className={cn(CLASSES.featuredComment.$root, styles.root)}
-        data-testid={`featuredComment-${comment.id}`}
+        data-testid={`commentAnswer-${comment.id}`}
+        aria-labelledby={`commentAnswerLabel-${comment.id}`}
+        id={`commentAnswer-${comment.id}`}
       >
+        <Hidden id={`commentAnswerLabel-${comment.id}`}>
+          Answer from {comment.author?.username || "Deleted User"} {` `}
+          <RelativeTime date={comment.createdAt} />
+        </Hidden>
```
Comment (cvle) [inline src/core/client/stream/tabs/Comments/Stream/AnsweredCommentsTab/AnsweredCommentContainer.tsx:86]: This

**Review by nick-funk — APPROVED** [review]
Works great!

**Pull request merged**